dependency-analysis/main.go

```diff
@@ -0,0 +1,357 @@
+	sb.WriteString("<th>AdvisoryUrl</th>\n")
+	sb.WriteString("</tr>\n")
+	for _, v := range i.Vulnerabilities {
+		sb.WriteString("<tr>\n")
+		sb.WriteString(fmt.Sprintf("<td>%s</td>\n", v.Severity))
+		sb.WriteString(fmt.Sprintf("<td>%s</td>\n", v.AdvisoryGHSAId))
+		sb.WriteString(fmt.Sprintf("<td>%s</td>\n", v.AdvisorySummary))
+		sb.WriteString(fmt.Sprintf("<td>%s</td>\n", v.AdvisoryURL))
+	}
+	sb.WriteString("</table>\n")
+	sb.WriteString("</details>\n")
+	vulnerabilities += sb.String()
+	return vulnerabilities
+}
+
+// getDefaultBranch gets the default branch of the repository.
+func getDefaultBranch(owner, repo string, client *github.Client) (string, error) {
+	ctx := context.Background()
+	repository, _, err := client.Repositories.Get(ctx, owner, repo)
+	if err != nil {
+		return "", fmt.Errorf("failed to get repository: %w", err)
+	}
+
+	return repository.GetDefaultBranch(), nil
+}
+
+// Validate validates the input parameters.
+func Validate(token, repo, commitSHA, pr string) error {
+	if token == "" {
+		return fmt.Errorf("token is empty") //nolint:goerr113
+	}
+	if repo == "" {
+		return fmt.Errorf("repo is empty") //nolint:goerr113
+	}
+	if commitSHA == "" {
+		return fmt.Errorf("commitSHA is empty") //nolint:goerr113
+	}
+	if pr == "" {
+		return fmt.Errorf("pr is empty") //nolint:goerr113
+	}
+	return nil
+}
+
+// createOrUpdateComment creates a new comment on the pull request or updates an existing one.
+func createOrUpdateComment(client *github.Client, owner, githubUser, repo string, prNum int, commentBody string) error {
+	comments, _, err := client.Issues.ListComments(
+		context.Background(), owner, repo, prNum, &github.IssueListCommentsOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get comments: %w", err)
+	}
+	// Check if the user has already left a comment on the pull request.
+	var existingComment *github.IssueComment
+	for _, comment := range comments {
+		if comment.GetUser().GetLogin() == githubUser {
+			existingComment = comment
+			break
+		}
+	}
+
+	// If the user has already left a comment, update it.
+	if existingComment != nil {
+		existingComment.Body = &commentBody
+		_, _, err = client.Issues.EditComment(context.Background(), owner, repo, *existingComment.ID, existingComment)
+		if err != nil {
+			return fmt.Errorf("failed to update comment: %w", err)
+		}
+		log.Println("Comment updated successfully!")
+	} else {
+		// Otherwise, create a new comment.
+		newComment := &github.IssueComment{
+			Body: &commentBody,
+		}
+		_, _, err = client.Issues.CreateComment(context.Background(), owner, repo, prNum, newComment)
+		if err != nil {
+			return fmt.Errorf("failed to create comment: %w", err)
+		}
+		log.Println("Comment created successfully!")
+	}
+	return nil
+}
+
+// GitHubIssueComment returns a markdown string for a GitHub issue comment.
+func GitHubIssueComment(checks *ScorecardResult) string {
+	if checks.Repo.Name == "" {
+		return ""
+	}
+	sb := strings.Builder{}
```

Comment:
maybe create a template file rather than hardcoded this layout in code. It' pretty hard to read otherwise

Reply:
I agree. I can do that.